frontend/file/src/main/java/com/virtuslab/gitmachete/frontend/file/quickfix/CreateBranchQuickFix.java

```diff
@@ -0,0 +1,66 @@
+package com.virtuslab.gitmachete.frontend.file.quickfix;
+
+import static com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle.getNonHtmlString;
+
+import java.util.Collections;
+
+import com.intellij.codeInsight.intention.IntentionAction;
+import com.intellij.codeInspection.util.IntentionFamilyName;
+import com.intellij.codeInspection.util.IntentionName;
+import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.project.Project;
+import com.intellij.psi.PsiFile;
+import git4idea.branch.GitBrancher;
+import git4idea.repo.GitRepository;
+import lombok.RequiredArgsConstructor;
+import lombok.experimental.ExtensionMethod;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle;
+import com.virtuslab.qual.guieffect.IgnoreUIThreadUnsafeCalls;
+
+@RequiredArgsConstructor
+@ExtensionMethod({GitMacheteBundle.class})
+public class CreateBranchQuickFix implements IntentionAction {
+
+  private final String branch;
+  private final String parentBranch;
+  private final PsiFile macheteFile;
+  private final @Nullable GitRepository gitRepository;
+
+  @Override
+  public @IntentionName String getText() {
+    return getNonHtmlString("action.GitMachete.MacheteAnnotator.IntentionAction.create-nonexistent-branch")
+        .format(branch, parentBranch);
+  }
+
+  @Override
+  public @IntentionFamilyName String getFamilyName() {
+    return "Git Machete";
+  }
+
+  @Override
+  public boolean isAvailable(Project project, Editor editor, PsiFile file) {
+    return true;
+  }
+
+  @Override
+  public void invoke(Project project, Editor editor, PsiFile file) {
```

Comment:
Okay... this one can be marked as `@UIEffect` right? as you've checked experimentally that it's executed on UI thread right?

Reply:
It is executed on the Swing's event dispatch thread. Let me add this and. we'll see if that is OK.

Reply:
Ok, I can't add this annotation, since:
> /root/git-machete-intellij-plugin/frontend/file/src/main/java/com/virtuslab/gitmachete/frontend/file/quickfix/CreateBranchQuickFix.java:49: error: [[guieffect, allcheckers]:override.effect] A method override may only be @UI if it overrides an @UI method.